.evergreen/config.yml

```diff
@@ -741,7 +730,7 @@ functions:
         working_dir: src/go.mongodb.org/mongo-driver
         script: |
           ${PREPARE_SHELL}
-          if [ "${SKIP_ECS_AUTH_TEST}" == "true" ]; then
+          if [ "${SKIP_ECS_AUTH_TEST}" = "true" ]; then
```

Comment:
`[ "${SKIP_ECS_AUTH_TEST}" == "true" ]` was expanding to `[ == "true" ]` when `SKIP_ECS_AUTH_TEST` was undefined (and the test was meant to be run). `[ == "true" ]` is not valid `sh`, and an error was being thrown in test setup. It did not cause a crash, but I thought it'd be good to avoid the error.

Using a single equality sign means this statement resolves to `[ "" == "true" ]`, which is valid `sh`. I'm actually unsure why this works the way it does. From my understanding, `==` and `=` are the same within single brackets in `sh`. It might be something to do with the way evergreen resolves expansions.

Note that replacing this with `[[ "{SKIP_ECS_AUTH_TEST}" == "true" ]]` does work as well.

Reply:
Thanks for the explanation! I also verified in the linked patch build that Ubuntu runs all tests, Windows skips the ECS test, and macos skips both ECS and EC2 tests.

Reply:
Huh, that is funky behavior. I would expect `[ "${SKIP_ECS_AUTH_TEST}" == "true" ]` to become `[ "" == "true" ]` after evergreen expansions are substituted and before the shell processes it.

I thought this could be due to [`dash` being the shell used on Debian](https://github.com/evergreen-ci/evergreen/wiki/Project-Commands#shellexec). I have dash on my mac, and running

```
if [ "" == "true" ]; then
    echo "foo"
else
    echo "bar"
fi
```
With `/bin/dash` gives an error. `test.sh: 1: [: unexpected operator`.

But it looks like these were run on Ubuntu 18.04. So... not sure what was going on there.

Reply:
Interesting... I suppose it's possible that bash in Ubuntu 18.04 has the same behavior has dash in Debian 🤷 . In any case, I see no other instances of `==` in our Evergreen config, so hopefully this won't be a problem in the future.